plugins/extra/forecast.go

```diff
@@ -50,17 +50,18 @@ type forecastPlugin struct {
 // geocoded location, along with the user who requested this be their home
 // location.
 type ForecastLocation struct {
-	Nick    string
+	ID      int64
+	Nick    string `xorm:"unique"`
 	Address string
 	Lat     float64
 	Lon     float64
 }
 
-func newForecastPlugin(b *seabird.Bot, cm *seabird.CommandMux, db *nut.DB) error {
+func newForecastPlugin(b *seabird.Bot, cm *seabird.CommandMux, db *xorm.Engine) error {
 	p := &forecastPlugin{db: db}
 
-	// Ensure the table is created if it doesn't exist
-	err := p.db.EnsureBucket("forecast_location")
+	// Ensure DB tables are up to date
+	err := p.db.Sync(ForecastLocation{})
```

Comment:
This is neato

Reply:
The thing I like about it as well is that it doesn't remove columns and it will perform very specific operations. It's also possible to pass it a partial model struct so you can write a simple migration framework around it - it's how gitea and gogs handle migrations.